admin_frontend/src/web_api.rs

```diff
@@ -21,23 +25,77 @@ pub fn router() -> Router<AppState> {
     .route("/login", post(login_handler))
     .route("/login_refresh/:refresh_token", post(login_refresh_handler))
     .route("/logout", post(logout_handler))
+
+    // user
+    .route("/change_password", post(change_password_handler))
+    .route("/oauth_login/:provider", post(post_oauth_login_handler))
+    .route("/invite", post(invite_handler))
+    .route("/open_app", post(open_app_handler))
+
+    // admin
     .route("/admin/user", post(admin_add_user_handler))
     .route(
-      "/admin/user/:param",
+      "/admin/user/:user_uuid",
       delete(admin_delete_user_handler).put(admin_update_user_handler),
     )
     .route(
       "/admin/user/:email/generate-link",
       post(post_user_generate_link_handler),
     )
-    .route("/change_password", post(change_password_handler))
-    .route("/oauth_login/:provider", post(post_oauth_login_handler))
+}
+
+// provide a link which when open in browser, opens the appflowy app
+pub async fn open_app_handler(
+  State(state): State<AppState>,
+  session: UserSession,
+) -> Result<WebApiResponse<WebApiOpenAppResponse>, WebApiError<'static>> {
+  let access_token_resp = state
+    .gotrue_client
+    .token(&Grant::RefreshToken(RefreshTokenGrant {
+      refresh_token: session.refresh_token.to_owned(),
+    }))
+    .await?;
+
+  let app_sign_in_url = format!(
+      "appflowy-flutter://login-callback#access_token={}&expires_at={}&expires_in={}&refresh_token={}&token_type={}",
```

Comment:
The `appflowy-flutter://login-callback` requires documentation to explain its usage. One issue to note is that the URL will become invalid if we alter the schema. For now, we can proceed with it.

Reply:
I'll add some comments to clarify this part